src/module/data/pseudo-documents/pseudo-document.mjs

```diff
@@ -233,6 +235,38 @@ export default class PseudoDocument extends foundry.abstract.DataModel {
     };
   }
 
+  /* -------------------------------------------------- */
+
+  /**
+   * A helper function to handle obtaining the relevant PseudoDocument from dropped data provided via a DataTransfer event.
+   * The dropped data could have:
+   * 1. A data object explicitly provided
+   * 2. A UUID.
+   *
+   * @param {object} data           The data object extracted from a DataTransfer event.
+   * @returns {Promise<PseudoDocument>}   The resolved PseudoDocument.
+   * @throws If a Document could not be retrieved from the provided data.
+   */
+  static async fromDropData(data) {
+    let pseudo = null;
+
+    // Case 1 - Data explicitly provided
+    if (data.data) pseudo = new this(data.data);
```

Comment:
I understand we align with core here, but it doesn't seem necessary. There will always be a `uuid`.